config/keys.go

```diff
@@ -65,5 +194,74 @@ func LoadSigningKeys() (*SigningKeys, error) {
 		}
 		return nil, err
 	}
+
+	if err := validateKeys(&config); err != nil {
+		return nil, err
+	}
+
 	return &config, nil
 }
+
+// LoadExecSaveSigningKeys loads signing key, executes given function and then saves the signing key
+func LoadExecSaveSigningKeys(fn func(keys *SigningKeys) error) error {
+	// core process
+	signingKeys, err := LoadSigningKeys()
+	if err != nil {
+		return err
+	}
+
+	if err := fn(signingKeys); err != nil {
+		return err
+	}
+
+	if err := signingKeys.Save(); err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
nit: code can be simplified
```suggestion
	return signingKeys.Save()
```